api/handlers.go

```diff
@@ -526,26 +526,29 @@ func getContainerJSON(c *context, w http.ResponseWriter, r *http.Request) {
 func postContainersCreate(c *context, w http.ResponseWriter, r *http.Request) {
 	r.ParseForm()
 	var (
-		config = dockerclient.ContainerConfig{
-			HostConfig: dockerclient.HostConfig{
-				MemorySwappiness: -1,
+		defaultMemorySwappiness = int64(-1)
+		name                    = r.Form.Get("name")
+		config                  = cluster.ContainerConfig{
+			HostConfig: containertypes.HostConfig{
+				Resources: containertypes.Resources{
+					MemorySwappiness: &(defaultMemorySwappiness),
+				},
 			},
 		}
-		name = r.Form.Get("name")
 	)
 
 	if err := json.NewDecoder(r.Body).Decode(&config); err != nil {
 		httpError(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 	// Pass auth information along if present
-	var authConfig *dockerclient.AuthConfig
+	var authConfig *apitypes.AuthConfig
 	buf, err := base64.URLEncoding.DecodeString(r.Header.Get("X-Registry-Auth"))
 	if err == nil {
-		authConfig = &dockerclient.AuthConfig{}
+		authConfig = &apitypes.AuthConfig{}
 		json.Unmarshal(buf, authConfig)
 	}
-	containerConfig := cluster.BuildContainerConfig(config)
+	containerConfig := cluster.BuildContainerConfig(config.Config, config.HostConfig, config.NetworkingConfig)
```

Comment:
I think it made sense earlier when the config struct looked different compared to what it is right now. I should be possible to pass a single config object now. I'm wondering if I should fix that in a follow-up PR though, since the current one will heavily change because of this.